internal/resources/azure/sql_managed_instance.go

```diff
@@ -0,0 +1,196 @@
+			PriceFilter: priceFilterConsumption,
+		},
+	}
+
+	if r.BackupStorageGb != nil {
+		costComponents = append(costComponents, r.sqlMIStorageCostComponent(), r.sqlMIBackupCostComponent())
+	}
+
+	if r.LicenceType == "LicenseIncluded" {
+		costComponents = append(costComponents, r.sqlMILicenseCostComponent())
+	}
+	if r.LongTermRetentionStorageGB != nil {
+		costComponents = append(costComponents, r.sqlMILongTermRetentionStorageGBCostComponent())
+	}
+	return costComponents
+}
+
+func (r *SQLManagedInstance) productDescription() *string {
+
+	productDescription := ""
+	if strings.Contains(r.SKU, "GP") {
+		productDescription = "SQL Managed Instance General Purpose"
+	} else if strings.Contains(r.SKU, "BC") {
+		productDescription = "SQL Managed Instance Business Critical"
+	}
+
+	if strings.Contains(r.SKU, "Gen5") {
+		productDescription = fmt.Sprintf("%s - %s", productDescription, "Compute Gen5")
+	}
+	return strPtr(productDescription)
+}
+
+func (r *SQLManagedInstance) meteredName() *string {
+	meterName := fmt.Sprintf("%s %s", strconv.FormatInt(*r.Cores, 10), "vCore")
+	return strPtr(meterName)
+}
+
+func (r *SQLManagedInstance) sqlMIStorageCostComponent() *schema.CostComponent {
+
+	return &schema.CostComponent{
+		Name:            fmt.Sprintf("Storage %s Gb (first 32 Gb include)", strings.ToTitle(strconv.FormatInt(*r.StorageSizeInGb, 10))),
+		Unit:            "Unit of 32Gb",
+		UnitMultiplier:  decimal.NewFromInt(1),
+		MonthlyQuantity: decimalPtr(decimal.NewFromInt(*r.StorageSizeInGb - 32)),
+		ProductFilter: &schema.ProductFilter{
+			VendorName:    strPtr(vendorName),
+			Region:        strPtr(r.Region),
+			Service:       strPtr(sqlMIServiceName),
+			ProductFamily: strPtr(sqlMIProductFamily),
+			AttributeFilters: ([]*schema.AttributeFilter{
+				{Key: "productName", Value: strPtr("SQL Managed Instance General Purpose - Storage")},
+				{Key: "meterName", Value: strPtr("Data Stored")},
+			}),
+		},
+		PriceFilter: priceFilterConsumption,
+	}
+}
+
+func (r *SQLManagedInstance) sqlMIBackupCostComponent() *schema.CostComponent {
+	backupCostComponent := schema.CostComponent{}
+	backupCostComponent.Name = fmt.Sprintf("Backup Cost for %s Gb (type %s)", strconv.FormatInt(*r.BackupStorageGb, 10), r.StorageAccountType)
+	backupCostComponent.Unit = "Gb"
+	backupCostComponent.UnitMultiplier = decimal.NewFromInt(1)
+	backupCostComponent.MonthlyQuantity = decimalPtr(decimal.NewFromInt(*r.BackupStorageGb))
+	backupCostComponent.ProductFilter = &schema.ProductFilter{
+		VendorName:    strPtr(vendorName),
+		Region:        strPtr(r.Region),
+		Service:       strPtr(sqlMIServiceName),
+		ProductFamily: strPtr(sqlMIProductFamily),
+		AttributeFilters: ([]*schema.AttributeFilter{
+			{Key: "productName", Value: strPtr("SQL Managed Instance PITR Backup Storage")},
+			{Key: "meterName", Value: strPtr(fmt.Sprintf("%s Data Stored", r.StorageAccountType))},
+		}),
+	}
+	backupCostComponent.PriceFilter = priceFilterConsumption
+	return &backupCostComponent
```

Comment:
could we change this to be a literal definition - as the other cost components are.